### PR TITLE
docs: update README, add release process

### PR DESCRIPTION
**Pull request by DaniPopes**
- Fill README with more stuff. [Rendered](https://github.com/ithacaxyz/solar/blob/dani/readme-update/README.md).
- Add [`cargo-release`](https://github.com/crate-ci/cargo-release) + [`dist`](https://github.com/axodotdev/cargo-dist) as the release framework.
- Clean up benchmark README.

### CODE REVIEW

File: RELEASE_CHECKLIST.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+- [ ] Create a new branch: `git checkout -b release-<version>`
+- [ ] Update CHANGELOG.md with the new version and the changes since the last release.
+- [ ] Run `cargo-release` to handle the version bump and commit: `cargo release --execute --no-publish --no-tag --allow-branch=<branch> <version>`
+- [ ] Push, open and merge the PR. The name of the PR should be the same as the `cargo-release` commit message.
+- [ ] `git checkout main` and `git pull`.
+- [ ] Verify `dist plan` is correct.
+- [ ] Run `cargo-release` to tag and publish: `cargo release --execute [--no-verify] <version>`
+
+These steps are adapted from the recommended `cargo-release` + `dist` workflow, described in more detail [here](https://opensource.axo.dev/cargo-dist/book/workspaces/cargo-release-guide.html#using-cargo-release-with-pull-requests).
```
Comment (mattsse) [inline RELEASE_CHECKLIST.md:24]: ah okay, so the pr is created first then dist then the actual release

File: crates/solar/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "solar-compiler"
-description = "Solar Solidity compiler"
+description = "Blazingly fast Solidity compiler"
```
Comment (mattsse) [inline crates/solar/Cargo.toml:3]: hehe

File: dist-workspace.toml
Comment (mattsse) [inline dist-workspace.toml:1]: this looks interesting
could be very useful for foundry potentially

File: RELEASE_CHECKLIST.md (hunk 1 above)
Comment (DaniPopes) [reply RELEASE_CHECKLIST.md:24]: the dist command is just to verify, dist runs on pushed tag event from the auto generated workflow

File: dist-workspace.toml
Comment (DaniPopes) [reply dist-workspace.toml:1]: Definitely, would have to figure out nightlies first / triggering releases automatically with an auto generated tag

File: README.md [hunk 3]
```diff
@@ -1,12 +1,134 @@
...
-[![Apache-2.0 License][apache-badge]][apache-url]
-[![CI Status][actions-badge]][actions-url]
+[![Crates.io](https://img.shields.io/crates/v/solar-compiler.svg)](https://crates.io/crates/solar-compiler)
+[![Downloads](https://img.shields.io/crates/d/solar-compiler)](https://crates.io/crates/solar-compiler)
+[![MIT License](https://img.shields.io/badge/license-MIT-blue.svg)](/LICENSE-MIT)
+[![Apache-2.0 License](https://img.shields.io/badge/license-Apache--2.0-blue.svg)](/LICENSE-APACHE)
+[![Actions status](https://github.com/ithacaxyz/solar/workflows/CI/badge.svg)](https://github.com/ithacaxyz/solar/actions)
 
-Rust implementation of the Solidity language.
+Blazingly fast, modular and contributor friendly Solidity compiler, written in Rust.
```
Comment (jenpaff) [inline README.md:9]: what do you think of adding some benchmarks?

File: README.md [hunk 4]
```diff
@@ -1,12 +1,134 @@
...
+
+## Features and Goals
+
+- Instant compiles and low memory usage
+- Expressive and useful diagnostics
+- Modular, library based architecture
+- Simple and hackable code base
+- Compatibility with the latest Solidity language breaking version (0.8.*)
+
+## Roadmap
```
Comment (jenpaff) [inline README.md]: just a small nit: Think the roadmap can go further down so beginning can just focus on what is it > why use it > getting started.

File: README.md [hunk 5]
```diff
@@ -1,12 +1,134 @@
...
+solar = { version = "0.1.0", package = "solar-compiler" }
+```
+
+Or through the CLI:
+
+```console
+$ cargo add solar-compiler --rename solar
+```
+
+You can see examples of how to use Solar as a library in the [examples](/examples) directory.
```
Comment (jenpaff) [inline README.md:44]: Like the idea of adding examples a lot! Do you have any more ideas what to add to the examples to give more direction ?

File: RELEASE_CHECKLIST.md (hunk 1 above)
Comment (jenpaff) [reply RELEASE_CHECKLIST.md:24]: sounds like a good release flow !

File: README.md (hunk 5 above)
Comment (DaniPopes) [reply README.md:44]: will have more ideas once we integrate more in foundry

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,390 @@
...
+
+This guide will help you get started. **Do not let this guide intimidate you**.
+It should be considered a map to help you navigate the process.
+
+## Conduct
+
+The Solar project adheres to the [Rust Code of Conduct][coc]. This describes
+the _minimum_ behavior expected from all contributors. Instances of violations of the
+Code of Conduct can be reported by contacting the project team at
+[moderation@Solar.rs](mailto:moderation@Solar.rs).
```
Comment (jenpaff) [inline CONTRIBUTING.md:20]: sent a test mail- lmk if it worked

File: benches/README.md [hunk 7]
```diff
@@ -3,9 +3,11 @@
 Simple benchmarks across different Solidity parser implementations.
 
 Run with:
+
 ```bash
 # Criterion
-cargo criterion -p solar-bench --bench bench
+cargo criterion -p solar-bench --bench bench -- --quiet --format terse |& tee benches/tables.in
```
Comment (jenpaff) [inline benches/README.md:11]: i'm getting an error when i run the command? am I missing something? 

```
➜  benches git:(dani/readme-update) ✗ cargo criterion -p solar-bench --bench bench -- --quiet --format terse |& tee benches/tables.in
   Compiling solar-bench v0.1.0 (/Users/jenpaff/work/solar/benches)
error: couldn't read `benches/src/../../testdata/solidity/test/benchmarks/verifier.sol`: No such file or directory (os error 2)
  --> benches/src/lib.rs:8:33
   |
8  |         source_from_path($path, include_str!($path))
   |                                 ^^^^^^^^^^^^^^^^^^^
...
18 |             include_source!("../../testdata/solidity/test/benchmarks/verifier.sol"),
   |             ----------------------------------------------------------------------- in this macro invocation
   |
   = note: this error originates in the macro `include_str` which comes from the expansion of the macro `include_source` (in Nightly builds, run with -Z macro-backtrace for more info)

error: couldn't read `benches/src/../../testdata/solidity/test/benchmarks/OptimizorClub.sol`: No such file or directory (os error 2)
  --> benches/src/lib.rs:8:33
   |
8  |         source_from_path($path, include_str!($path))
   |                                 ^^^^^^^^^^^^^^^^^^^
...
19 |             include_source!("../../testdata/solidity/test/benchmarks/OptimizorClub.sol"),
   |             ---------------------------------------------------------------------------- in this macro invocation
   |
   = note: this error originates in the macro `include_str` which comes from the expansion of the macro `include_source` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `solar-bench` (lib) due to 2 previous errors
Error: 'cargo bench' returned an error (exit status: 101); unable to continue.
```

Comment (DaniPopes) [reply benches/README.md:11]: need to clone submodules: `git submodule update --init`
will add to the comment

File: CONTRIBUTING.md (hunk 6 above)
Comment (DaniPopes) [reply CONTRIBUTING.md:20]: oh sorry forgot about that part after search+replace, i don't think we even have that domain

Should we add that mailbox to ithaca.xyz?

Comment (gakonst) [reply CONTRIBUTING.md:20]: I think just do gerogios@ithaca.xyz here

**Pull request merged**